zebra-consensus/src/transaction/check.rs

```diff
@@ -30,10 +32,21 @@ pub fn has_inputs_and_outputs(tx: &Transaction) -> Result<(), TransactionError>
     let n_spends_sapling = tx.sapling_spends_per_anchor().count();
     let n_outputs_sapling = tx.sapling_outputs().count();
     let n_actions_orchard = tx.orchard_actions().count();
+    let flags_orchard = tx.orchard_flags().unwrap_or_else(Flags::empty);
 
-    if tx_in_count + n_spends_sapling + n_joinsplit + n_actions_orchard == 0 {
+    if tx_in_count
+        + n_spends_sapling
+        + n_joinsplit
+        + (n_actions_orchard > 0 && flags_orchard.contains(Flags::ENABLE_SPENDS)) as usize
+        == 0
+    {
         Err(TransactionError::NoInputs)
-    } else if tx_out_count + n_outputs_sapling + n_joinsplit + n_actions_orchard == 0 {
+    } else if tx_out_count
+        + n_outputs_sapling
+        + n_joinsplit
+        + (n_actions_orchard > 0 && flags_orchard.contains(Flags::ENABLE_OUTPUTS)) as usize
```

Comment:
Confusing code: The actions lines are a bit long and confusing, can you split them out into their own variables?

Reply:
I created a ticket (https://github.com/ZcashFoundation/zebra/issues/2410) for this and added a TODO comment into the code in https://github.com/ZcashFoundation/zebra/pull/2398/commits/05019a9d4d5b7a18517489029f9a9a19171b76c4